### PR TITLE
added error messages to login, register and forget-password forms

### PR DESCRIPTION
**Pull request by anastasiyaSht**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/secure/RapidBoard.jspa?rapidView=777)

## Code reviewers

- [ ] @Sokolovskyi1995
- [ ] @BDubas
- [x] @rromanitc
- [ ] @martasakal
- [ ] @Zaplatynskyi
- [ ] @aberezh

### Second Level Review

- [ ] @lhalam
- [ ] @ssOleg

## Summary of issue

ToDo

In Register form: disabled 'Register' button, if input data is incorrect. If a mail has been already registered, the user will see a message 'This mail is already registered.'
In Login form: disabled 'Login' button, if input data is incorrect. If user input invalid data, he will see a message 'Invalid mail or password'
I Forget-Password form: disabled 'Submit' button if input data is incorrect. 
If a mail has not been registered yet, the user will see a message 'Invalid mail.'
User cannot put new password the same, as old.

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by BDubas — DISMISSED** [review]

**Review by Zaplatynskyi — DISMISSED** [review]

**Review by rromanitc — DISMISSED** [review]

**Review by martasakal — DISMISSED** [review]

**Review by rromanitc — DISMISSED** [review]

**Review by BDubas — DISMISSED** [review]

File: eventually/static/src/containers/registerLogin/Forget.js [hunk 1]
```diff
@@ -37,26 +40,61 @@ class Forget extends React.Component {
 
     handleSubmit = event => {
         const email = this.state.email;
-        forgetPasswordService(email);
+        forgetPasswordService(email).then((response) => {
+            if (response.status !== 200) {
+                this.setState({sentLetter:false});
```
Comment (lhalam) [inline eventually/static/src/containers/registerLogin/Forget.js]: marge to
```
 this.setState({sentLetter: false,
                messageError: true});
```

File: eventually/static/src/containers/registerLogin/Login.js [hunk 2]
```diff
@@ -22,9 +22,15 @@ class Login extends React.Component {
             password: '',
             MessageEmail: '',
             MessagePassword: '',
+            messageError: false,
         };
     }
 
+    componentDidCatch(error, info) {
+        console.log(info);
```
Comment (lhalam) [inline eventually/static/src/containers/registerLogin/Login.js]: remove `console.log`

**Review by lhalam — CHANGES_REQUESTED** [review]

File: eventually/static/src/containers/registerLogin/Forget.js [hunk 3]
```diff
@@ -37,26 +40,60 @@ class Forget extends React.Component {
 
     handleSubmit = event => {
         const email = this.state.email;
-        forgetPasswordService(email);
+        forgetPasswordService(email).then((response) => {
```
Comment (lhalam) [inline eventually/static/src/containers/registerLogin/Forget.js:43]: use  .then(function (response) {
    console.log(response);
  })
  .catch(function (error) {
    console.log(error);
  });

File: eventually/static/src/containers/registerLogin/Login.js [hunk 4]
```diff
@@ -51,12 +52,26 @@ class Login extends React.Component {
         const password = this.state.password;
         loginService(email, password)
             .then((response) => {
```
Comment (lhalam) [inline eventually/static/src/containers/registerLogin/Login.js]: use .then(function (response) {
    console.log(response);
  })
  .catch(function (error) {
    console.log(error);
  });

File: eventually/static/src/containers/registerLogin/Forget.js [hunk 5]
```diff
@@ -37,26 +40,60 @@ class Forget extends React.Component {
...
+            if (response.status !== 200) {
+                this.setState({messageError:true});
```
Comment (lhalam) [inline eventually/static/src/containers/registerLogin/Forget.js]: use space` messageError: true`

**Review by rromanitc — DISMISSED** [review]

**Review by lhalam — DISMISSED** [review]

**Review by aberezh — APPROVED** [review]

**Review by BDubas — APPROVED** [review]

**Review by lhalam — APPROVED** [review]

**Review by rromanitc — APPROVED** [review]

**Pull request merged**